crates/datastore/src/locking_tx_datastore/datastore.rs

```diff
@@ -217,7 +217,8 @@ impl Locking {
         }
 
         // Fix up auto_inc IDs in the cached system table schemas.
-        committed_state.reset_system_table_schemas()?;
+        // We might need something to reset sequence ids?
```

Comment:
I think we should include an assertion here that the system table IDs match the ones we expect. If that assertion holds, then we don't need to `reset_system_table_schemas` AFAIK, and if it doesn't hold then this patch probably (?) breaks that snapshot.

Reply:
The call  to `reset_system_table_schemas` was previously necessary here for the same reason it was in the bootstrap path. When restoring from a snapshot, we don't respect the snapshot's system table schemas, instead we use the statically known schemas. (You can see this in action on lines 184 -- 190 of this file.) Prior to your change, said schemas were full of sentinel ID values, and so we had to reset them to use the updated IDs. With your change, the IDs should just be installed correctly already.

Reply:
I added a function which just asserts that the ids match, and I updated the comment.